eng/Version.Details.xml

```diff
@@ -25,8 +15,18 @@
       <Uri>https://github.com/dotnet/core-setup</Uri>
       <Sha>464f69912c1e644e1b6e3aa0184f5c7c1a690cec</Sha>
     </Dependency>
-  </ProductDependencies>
-  <ToolsetDependencies>
+    <Dependency Name="Microsoft.NETCore.Runtime.CoreCLR" Version="3.0.0-preview-27316-72">
```

Comment:
CoreCLR is indeed a product dependency right? We use it for our partial facades and as reference from some of our product assemblies. Unless the intent to use product dependency as “we ship these within our product”, is that the case?

Reply:
@ericstj Can you comment on this? We can move this one back to product if need be.

Reply:
@safern A better question here is this: If the exact version of coreclr mentioned here doesn't make it into the sdk/core-setup outputs, is there an issue?

Reply:
I think yes. Because we use CoreCLR to generate our facades and some projects use implementation details. So if we use a version that exposes and API and we use it in a project or generate a facade with that API, and then consume a previous CoreCLR version in core-setup that doesn’t contain that API I think it would be an issue, however I see that should be hard to happen.

Reply:
Okay I'm move that back for now.

Reply:
> If the exact version of coreclr mentioned here doesn't make it into the sdk/core-setup outputs, is there an issue?

**exact version** no, we just need a version that has the same public types exposed from CoreLib. 
 We have no *version* dependency nor *redistribution* of coreclr.  We only need one with the same public surface.

Historically we've allowed these repositories to build in parallel and I'd expect we'd want to do that moving forward.

Reply:
I think the values used in Version.Details.xml are kind of confusing, because `ProductDependency` could be a dependency needed in order to create your product. 

`TestDependency` means to me, something we use to test. Only core-setup is a test depedency in this case I believe. 

Coreclr would be more like `BuildDependency` and arcade `ToolChainDependency`.

Reply:
Or ReferenceDependency?  I really feel like we need a document with some clear definitions of these terms and the behavior that falls out of using them.

Reply:
It's maybe even more like "MustShipThisOrElseDependency" and "EverythingElseDependency"

Reply:
Yeah those would also clarify really from the devs mind what that dependency is intended for.